ingestion/src/metadata/ingestion/connections/test_connections.py

```diff
@@ -277,7 +277,7 @@ def test_connection_engine_step(connection: Engine) -> None:
     Generic step to validate the connection against a db
     """
     with connection.connect() as conn:
-        conn.execute(ConnTestFn())
+        conn.execute(ConnTestFn()).fetchall()
```

Comment:
why fetchall, if we really need to fetch lets do fetchone

Reply:
also if this is only required for synapse and not other connectors then can we move it in the connection file of synapse?

Reply:
yes will change to `fetchone`.
Had to change this as just using the `.execute` made the connection to be busy in some cases as it maybe sometimes returns a `generator` object leaving the connection object.
more info about it [here](https://stackoverflow.com/questions/46489749/whats-the-difference-between-using-c-fetchall-vs-just-assigning-c-execute)